### PR TITLE
Fix Android Studio 4.2 detection on Windows

### LINKED ISSUE #82345: Flutter Doctor cannot find Android Studio 4.2
In Android Studio 4.2, the ANDROID_SDK_HOME variable has been deprecated:

*Android Studio 4.2 release notes:*
flutter 

  On a clean install of AS and Flutter, `flutter doctor` cannot find Android Studio or JDK.

Initial error:
error

Temporary fix that (mostly) worked for me:
  1. Manually create JAVA_HOME system variable pointing to Android Studio JRE dir.
  2. Add AS JRE to PATH
  3. Do a `flutter config --android-studio-dir="path\to\studio\not\bin"` 
  *At this point, there will be a thread exception when trying to accept Android Licenses*
  4. Install "Android SDK command line tools" in SDK Manager

Finally, `flutter doctor` claims there are no issues.
success

### PR DESCRIPTION
**Pull request by TahaTesser**
### Before
```console
[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).
```

### After
```console
[√] Android Studio (version 4.2.0)
    • Android Studio at C:\Users\Taha\Code\android-studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)
```

Fixed https://github.com/flutter/flutter/issues/82345

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
LGTM!

**Comment by jonahwilliams** [comment]
Thank you @TahaTesser !

File: packages/flutter_tools/test/general.shard/android/android_studio_test.dart [hunk 1]
```diff
@@ -408,6 +408,24 @@ void main() {
     FileSystem: () => windowsFileSystem,
     ProcessManager: () => FakeProcessManager.any(),
   });
+
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/android/android_studio_test.dart:411]:
```suggestion
 
```

Comment (TahaTesser) [reply packages/flutter_tools/test/general.shard/android/android_studio_test.dart]: done

**Pull request merged**